server/public/model/post.go

```diff
@@ -657,6 +664,152 @@ func (o *Post) GetProp(key string) any {
 	return o.Props[key]
 }
 
+// ValidateProps checks all known props for validity.
+// Currently, it logs warnings for invalid props rather than returning an error.
+// In a future version, this will be updated to return errors for invalid props.
+func (o *Post) ValidateProps(logger mlog.LoggerIFace) {
+	if err := o.propsIsValid(); err != nil {
+		logger.Warn(
+			"Invalid post props. In a future version this will result in an error. Please update your integration to be compliant.",
+			mlog.String("post_id", o.Id),
+			mlog.Err(err),
+		)
+	}
+
+	for i, a := range o.Attachments() {
+		if err := a.IsValid(); err != nil {
+			logger.Warn(
+				"Invalid messsage attachment. In a future version this will result in an error. Please update your integration to be compliant.",
+				mlog.String("post_id", o.Id),
+				mlog.Int("attachment_index", i),
+				mlog.Err(err))
+		}
+	}
+}
+
+func (o *Post) propsIsValid() error {
+	var multiErr *multierror.Error
+
+	props := o.GetProps()
+
+	// Check basic props validity
+	if props == nil {
+		return nil
+	}
+
+	if props[PostPropsAddedUserId] != nil {
+		if addedUserID, ok := props[PostPropsAddedUserId].(string); !ok {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("added_user_id prop must be a string"))
+		} else if !IsValidId(addedUserID) {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("added_user_id prop must be a valid user ID"))
+		}
+	}
+	if props[PostPropsDeleteBy] != nil {
+		if deleteByID, ok := props[PostPropsDeleteBy].(string); !ok {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("delete_by prop must be a string"))
+		} else if !IsValidId(deleteByID) {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("delete_by prop must be a valid user ID"))
+		}
+	}
+
+	// Validate integration props
+	if props[PostPropsOverrideIconURL] != nil {
+		if iconURL, ok := props[PostPropsOverrideIconURL].(string); !ok {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("override_icon_url prop must be a string"))
+		} else if iconURL == "" || !IsValidHTTPURL(iconURL) {
+			multiErr = multierror.Append(multiErr, fmt.Errorf("override_icon_url prop must be a valid URL"))
+		}
+	}
+	if props[PostPropsOverrideIconEmoji] != nil {
```

Comment:
Didn't know this prop existed :thinking:

Reply:
I vaguely remember it. But I don't even know if it still works.